maven-core/src/main/java/org/apache/maven/lifecycle/providers/CleanLifecycleProvider.java

```diff
@@ -0,0 +1,61 @@
+package org.apache.maven.lifecycle.providers;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.maven.lifecycle.Lifecycle;
+import org.apache.maven.lifecycle.mapping.LifecyclePhase;
+
+@Named( "clean" )
+@Singleton
+public final class CleanLifecycleProvider
+    implements Provider<Lifecycle>
+{
+  private final Lifecycle lifecycle;
+
+  @Inject
+  public CleanLifecycleProvider()
+  {
+    this.lifecycle = new Lifecycle(
+        "clean",
+        ImmutableList.of(
+            "pre-clean",
+            "clean",
+            "post-clean"
+        ),
+        ImmutableMap.of(
```

Comment:
Can we use Java Collections API here instead of Guava?

Reply:
Does Java 8 has something so simple as that above? Am fine with whatever here, this PR was just done quickly to make @slachiewicz send rockets :smile:

Reply:
with Guava it will be easier to backport it to 3.x line. Thx @cstamas for so quick solution :)

Reply:
I don't think this should be on 3.8.x, at most 3.9.x. `Arrays.asList()` and `Collections.unmodifiableList()` should do it, no?

Reply:
Agreed, this is not for 3.8.x (well, unsure why to spend time back-porting it there).

For lists, yes, but what about maps? IMO, using pure Java is okay, but will be more bloated... anyway, what is the guava status in maven core? (as I searched codebase on master and did not find many uses of it).... so after this I think yes, we should omit use of Guava...

Reply:
backport to have more "Plexus free" Maven core so if in few years someone will open discussion about the minimal supported version we will have a cleaner situation. Ofc there is still maven-compat with its own challenges.

Reply:
@cstamas Correct, limit the use, at best be a transitive dep.

Reply:
done